CDS/WebContent/js/contest.js

```diff
@@ -1,284 +1,216 @@
 	}
 
-	var loadScoreboard = function() {
-		if (scoreboard != null)
+	loadAwards() {
+		if (this.awards != null)
 			return new $.Deferred().resolve();
 
-		return loadObject('scoreboard', function(result) { scoreboard = result });
-	}
-
-	var getContestURL = function() {
-		return contestURL;
-	}
-	var getInfo = function() {
-		return info;
-	}
-	var getState = function() {
-		return state;
-	}
-	var getLanguages = function() {
-		return languages;
-	}
-	var getJudgementTypes = function() {
-		return judgementTypes;
-	}
-	var getProblems = function() {
-		return problems;
-	}
-	var getSubmissions = function() {
-		return submissions;
-	}
-	var getJudgements = function() {
-		return judgements;
-	}
-	var getRuns = function() {
-		return runs;
-	}
-	var getClarifications = function() {
-		return clarifications;
-	}
-	var getScoreboard = function() {
-		return scoreboard;
-	}
-	var getOrganizations = function() {
-		return organizations;
-	}
-	var getGroups = function() {
-		return groups;
-	}
-	var getAwards = function() {
-		return awards;
-	}
-	var getStartStatus = function() {
-		return startStatus;
-	}
-	var getTeams = function() {
-		return teams;
-	}
-	var getTeamById = function(id) {
-		for (var i = 0; i < teams.length; i++) {
-			if (teams[i].id == id)
-				return teams[i];
-		}
-		return null;
-	}
-	var clear = function() {
-		startStatus = null;
-		problems = null;
-		submissions = null;
-		judgements = null;
-		clarifications = null;
-	}
-
-    var post = function(type, body, ok, fail) {
-        console.log("Posting (POST) contest object: " + type);
+		return this.loadObject('awards', (result) => { this.awards = result });
+	}
+
+	getContestURL() { return this.contestURL }
+	getInfo() { return this.info }
+	getState() { return this.state }
+	getStartStatus() { return this.startStatus }
+	getLanguages() { return this.languages }
+	getJudgementTypes() { return this.judgementTypes }
+	getProblems() { return this.problems }
+	getGroups() { return this.groups }
+	getTeams() { return this.teams }
+	getOrganizations() { return this.organizations }
+	getSubmissions() { return this.submissions }
+	getJudgements() { return this.judgements }
+	getRuns() { return this.runs }
+	getClarifications() { return this.clarifications }
+	getScoreboard() { return this.scoreboard }
+	getAwards() { return this.awards }
+
+	clear() {
+		this.startStatus = null;
+		this.problems = null;
+		this.submissions = null;
+		this.judgements = null;
+		this.clarifications = null;
+	}
+
+	post(type, body, ok, fail) {
+        console.log("POSTing contest object: " + type);
         return $.ajax({
-		    url: getURL(type),
+		    url: this.getURL(type),
 		    method: 'POST',
 		    headers: { "Accept": "application/json" },
 		    data: body,
-		    success: function(body) {
-		    	ok(body);
-		    },
-		    error: function(result) {
-			    fail(result);
-		    }
+		    success: ok(result),
```

Comment:
I don't think this works? Or does it? Since it doesn't know what `result` is here.

I'd expect one of:
```suggestion
		    success: (result) => ok(result),
```
or
```suggestion
		    success: ok,
```

Reply:
As discussed, will fix next PR